src/systemTest/java/com/flightstats/hub/system/functional/WebhookErrorTest.java

```diff
@@ -0,0 +1,156 @@
+package com.flightstats.hub.system.functional;
+
+import com.flightstats.hub.kubernetes.HubLifecycle;
+import com.flightstats.hub.model.Webhook;
+import javax.inject.Inject;
+
+import com.flightstats.hub.model.WebhookCallbackSetting;
+import com.flightstats.hub.system.ModelBuilder;
+import com.flightstats.hub.system.config.DependencyInjector;
+import com.flightstats.hub.system.service.CallbackService;
+import com.flightstats.hub.system.service.ChannelService;
+import com.flightstats.hub.system.service.WebhookService;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.RepeatedTest;
+
+import java.util.Comparator;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Slf4j
+//@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class WebhookErrorTest extends DependencyInjector {
+    @Inject
+    private ChannelService channelService;
+    @Inject
+    private WebhookService webhookService;
+    @Inject
+    private CallbackService callbackService;
+    @Inject
+    private HubLifecycle hubLifecycle;
+    @Inject
+    private ModelBuilder modelBuilder;
+
+    private Webhook webhook;
+    private String channelName;
+    private String webhookName;
+
+    @BeforeAll
+    void hubSetup() {
+        hubLifecycle.setup();
+    }
+
+    @BeforeEach
+    void before() {
+        this.channelName = randomAlphaNumeric(10);
+        this.webhookName = randomAlphaNumeric(10);
+
+        initChannelAndWebhook();
+    }
+
+    private Webhook buildWebhook() {
+        return modelBuilder.webhookBuilder()
+                .channelName(channelName)
+                .webhookName(webhookName)
+                .build();
+    }
+
+    private void initChannelAndWebhook() {
+        channelService.create(channelName);
+
+        createWebhook();
+    }
+
+    private void createWebhook() {
+        webhook = buildWebhook().withParallelCalls(1).withMaxAttempts(0);
+        webhookService.insertAndVerify(webhook);
+    }
+
+    private void verifyHasReceivedErrorForItem(String firstUrl) {
+        callbackService.awaitHubHasCallbackErrorForItemPath(webhookName, firstUrl);
+        assertTrue(callbackService.hasCallbackErrorInHub(webhookName, firstUrl));
+        log.info("Found callback error for first item {} {}", webhookName, firstUrl);
+    }
+
+    @RepeatedTest(3)
```

Comment:
is this intended?

Reply:
Yep! It's the way Junit 5 does the manual loop this file had been doing for its tests.

Reply:
Looks like it doesn't behave quite like I expect. I'll see if I can fix this up. Thanks for catching it.